coreneuron/apps/main1.cpp

```diff
@@ -601,6 +601,16 @@ extern "C" int run_solve_core(int argc, char** argv) {
         output_spikes(output_dir.c_str(), spikes_population_name);
     }
 
+    // copy weights back to NEURON NetCon
+    if (nrn2core_all_weights_return_) {
+        std::vector<double*> weights;
```

Comment:
Simple optimization:
```cpp
std::vector<double*> weights(nrn_nthread, NULL);
for (int i=0; i < nrn_nthread; ++i) {
          weights[i]=nrn_threads[i].weights;
}
```

Reply:
I agree. Thanks.  I have been explicitly casting int to std::size_t for indexing into std::vector. Is that uselessly verbose?